File: polemica-fantasy-admin/src/api/tournamentRequests.ts
import type { TournamentKind, TournamentStatus } from './types'

export interface StreamLinkRequest {
  label?: string | null
  url: string
}

export interface CreateTournamentRequest {
  name: string
  description?: string | null
  status: TournamentStatus
  kind?: TournamentKind | null
  polemicaCompetitionId?: number | null
  defaultExpectedGameCount?: number | null
  streamLinks?: StreamLinkRequest[]
}

export interface UpdateTournamentRequest {
  name?: string | null
  description?: string | null
  status?: TournamentStatus | null
  kind?: TournamentKind | null
  polemicaCompetitionId?: number | null
  defaultExpectedGameCount?: number | null
  streamLinks?: StreamLinkRequest[] | null
}

export interface AddTournamentPlayerRequest {
  fantasyPlayerId?: number | null
  polemicaUserId?: number | null
  nickname?: string | null
}

export interface PatchTournamentPlayerRequest {
  excludedFromPackPool: boolean
}

File: polemica-fantasy-backend/src/main/kotlin/io/github/mralex1810/fantasy/dto/admin/request/TournamentRequests.kt
package io.github.mralex1810.fantasy.dto.admin.request

import io.github.mralex1810.fantasy.entity.TournamentKind
import io.github.mralex1810.fantasy.entity.TournamentStatus
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size

data class CreateTournamentRequest(
    @field:NotBlank @field:Size(max = 512)
    val name: String,
    @field:Size(max = 10000)
    val description: String? = null,
    @field:NotNull
    val status: TournamentStatus = TournamentStatus.DRAFT,
    /** Defaults to STANDALONE when omitted in JSON. */
    val kind: TournamentKind? = null,
    val polemicaCompetitionId: Long? = null,
)

data class UpdateTournamentRequest(
    @field:Size(max = 512)
    val name: String? = null,
    @field:Size(max = 10000)
    val description: String? = null,
    val status: TournamentStatus? = null,
    val kind: TournamentKind? = null,
    val polemicaCompetitionId: Long? = null,
)

data class AddTournamentPlayerRequest(
    val fantasyPlayerId: Long? = null,
    val polemicaUserId: Long? = null,
    @field:Size(max = 512)
    val nickname: String? = null,
)

data class PatchTournamentPlayerRequest(
    @field:NotNull
    val excludedFromPackPool: Boolean,
)
